.changeset/tough-socks-tease.md

```diff
@@ -2,4 +2,6 @@
 "@saleor/app-sdk": major
 ---
 
-Breaking change: Remove checking "domain" header from Saleor requests. It should be replaced with the "saleor-api-url" header.
+Breaking change: Remove checking "saleor-domain" header from Saleor requests. It should be replaced with the "saleor-api-url" header.
```

Comment:
But who should check saleor-api-url? app or sdk? If SDK, it's not a part of the changelog but something we should do

Reply:
Nope, this is something SDK does: we have dropped `saleor-domain` which was deprecated in favor of `saleor-api-url` header in our validation, thus making SDK incompatible with versions prior to 3.15. I've improved the description so that it's more clear.